[BUG] StratifiedGroupKFold import breaks pytest discovery

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->
Pytest discovery fails at `sktime/classification/tests/test_sklearn_compatability.py` when importing 'StratifiedGroupKFold' from 'sklearn.model_selection'

**To Reproduce**
<!--
Add a Minimal, Complete, and Verifiable example (for more details, see e.g. https://stackoverflow.com/help/mcve

If the code is too long, feel free to put it in a public gist and link it in the issue: https://gist.github.com
-->
pytest discovery
```python
conda run python /Users/user/.vscode/extensions/ms-python.python-2022.0.1814523869/pythonFiles/get_output_via_markers.py /Users/user/.vscode/extensions/ms-python.python-2022.0.1814523869/pythonFiles/testing_tools/run_adapter.py discover pytest -- --rootdir /Users/user/dev/sktime -s --cache-clear
```

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->
Expected not to get 
```python
sktime/classification/tests/test_sklearn_compatability.py:17: in <module>
    from sklearn.model_selection import (
E   ImportError: cannot import name 'StratifiedGroupKFold' from 'sklearn.model_selection'
```

**Additional context**
<!--
Add any other context about the problem here.
-->
`StratifiedGroupKFold` is introduced in the latest stable version of sklearn (1.0.2). Sktime requires version 0.24.0 or higher, but if I install 1.0.2 then other tests (testing sklearn version) fail.
Temporary solution: comment out the import and use of `StratifiedGroupKFold` in `sktime/classification/tests/test_sklearn_compatability.py`

**Versions**
main commit 88c08e92c058b9c83322646744ec0241e2dcaaf5
<details>


System:
    python: 3.8.10 (default, May 19 2021, 11:01:55)  [Clang 10.0.0 ]
executable: /Users/user/opt/anaconda3/envs/sktime-dev/bin/python
   machine: macOS-10.16-x86_64-i386-64bit

Python dependencies:
          pip: 21.1.3
   setuptools: 52.0.0.post20210125
      sklearn: 0.24.2
       sktime: 0.10.1
  statsmodels: 0.12.2
        numpy: 1.21.2
        scipy: 1.7.1
       pandas: 1.3.4
   matplotlib: 3.4.2
       joblib: 1.1.0
        numba: 0.52.0
     pmdarima: 1.8.2
      tsfresh: 0.18.0

</details>

<!-- Thanks for contributing! -->

--- a/sktime/classification/tests/test_sklearn_compatability.py
+++ b/sktime/classification/tests/test_sklearn_compatability.py
@@ -14,6 +14,8 @@ import pytest
 from sklearn.calibration import CalibratedClassifierCV
 from sklearn.ensemble import VotingClassifier
 from sklearn.experimental import enable_halving_search_cv  # noqa
+
+# StratifiedGroupKFold, removed because it is not available in sklearn 0.24
 from sklearn.model_selection import (
     GridSearchCV,
     GroupKFold,
@@ -27,7 +29,6 @@ from sklearn.model_selection import (
     RandomizedSearchCV,
     RepeatedKFold,
     ShuffleSplit,
-    StratifiedGroupKFold,
     StratifiedKFold,
     StratifiedShuffleSplit,
     TimeSeriesSplit,
@@ -43,6 +44,7 @@ DATA_ARGS = [
     {"return_numpy": True, "n_columns": 2},
     {"return_numpy": False, "n_columns": 2},
 ]
+# StratifiedGroupKFold(n_splits=2), , removed, not available in sklearn 0.24
 CROSS_VALIDATION_METHODS = [
     KFold(n_splits=2),
     RepeatedKFold(n_splits=2, n_repeats=2),
@@ -52,7 +54,6 @@ CROSS_VALIDATION_METHODS = [
     StratifiedKFold(n_splits=2),
     StratifiedShuffleSplit(n_splits=2, test_size=0.25),
     GroupKFold(n_splits=2),
-    StratifiedGroupKFold(n_splits=2),
     LeavePGroupsOut(n_groups=5),
     GroupShuffleSplit(n_splits=2, test_size=0.25),
     TimeSeriesSplit(n_splits=2),

